### PR TITLE
Metrics with micrometer

### PR DESCRIPTION
**Pull request by algattik**
Implemented metrics per ADR and feedback at https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/637.

Updated sample 04.3 to include a Prometheus showcase.

The structure of interfaces around Instrumentation is a bit unwieldy, but is constrained by the limitations of the IoC container around optional dependencies.

Only some ExecutorServices are instrumented currently for illustration.

### CODE REVIEW

**Review by marcgs — APPROVED** [review]

**Comment by marcgs** [comment]
Approving changes as this PR was completed in collaboration with @algattik , reviewing each other's changes

File: core/micrometer/src/main/java/org/eclipse/dataspaceconnector/micrometer/MicrometerExtension.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    public void initialize(ServiceExtensionContext context) {
+        MeterRegistry registry = Metrics.globalRegistry;
+        context.registerService(MeterRegistry.class, registry);
+
+        var enableMetrics = Boolean.valueOf(context.getSetting(ENABLE_METRICS, "true"));
+        var enableSystemMetrics = Boolean.valueOf(context.getSetting(ENABLE_SYSTEM_METRICS, "true"));
+        var enableOkHttpMetrics = Boolean.valueOf(context.getSetting(ENABLE_OKHTTP_METRICS, "true"));
+        var enableExecutorMetrics = Boolean.valueOf(context.getSetting(ENABLE_EXECUTOR_METRICS, "true"));
+
+        if (enableMetrics && enableSystemMetrics) {
```
Comment (cpeeyush) [inline core/micrometer/src/main/java/org/eclipse/dataspaceconnector/micrometer/MicrometerExtension.java:61]: why we always need enableMetrics ? is enableSystemMetrics/enableOkHttpMetrics/enableExecutorMetrics is not enough ?

File: extensions/http/jersey-micrometer/src/main/java/org/eclipse/dataspaceconnector/extension/jersey/JerseyMicrometerExtension.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public String name() {
+        return "Jersey Micrometer Metrics";
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var enableMetrics = Boolean.valueOf(context.getSetting(ENABLE_METRICS, "true"));
+        var enableJerseyMetrics = Boolean.valueOf(context.getSetting(ENABLE_JERSEY_METRICS, "true"));
+
+        if (enableMetrics && enableJerseyMetrics) {
```
Comment (cpeeyush) [inline extensions/http/jersey-micrometer/src/main/java/org/eclipse/dataspaceconnector/extension/jersey/JerseyMicrometerExtension.java:48]: same question here for enableMetrics, as above.

**Review by cpeeyush — APPROVED** [review]
Approved with few questions.

File: core/micrometer/src/main/java/org/eclipse/dataspaceconnector/micrometer/MicrometerExtension.java (hunk 1 above)
Comment (marcgs) [reply core/micrometer/src/main/java/org/eclipse/dataspaceconnector/micrometer/MicrometerExtension.java:61]: As discussed, leaving it as is.

**Pull request merged**